cryptocoins@pbojan/files/cryptocoins@pbojan/desklet.js

```diff
@@ -13,307 +13,338 @@ const DESKLET_ROOT = imports.ui.deskletManager.deskletMeta[UUID].path;
+      width: WIDTH_ICON,
+      style_class: 'containerLeft'
+    });
+    var file = Gio.file_new_for_path(DESKLET_ROOT + '/images/icons/' + data['symbol'].toLowerCase() + '.png');
+    var gicon = new Gio.FileIcon({file: file});
+    var image = new St.Icon({
+      gicon: gicon,
+      icon_size: WIDTH_ICON,
+      icon_type: St.IconType.SYMBOLIC,
+      style_class: 'icon'
+    });
+    left.add(image);
+
+    var right = new St.BoxLayout({
+      vertical: true,
+      width: WIDTH - PADDING - WIDTH_ICON,
+      style_class: 'containerRight'
+    });
+    var label = new St.Label({
+      style_class: 'header'
+    });
+    label.set_text(data['name']);
+    right.add(label);
+
+    label = new St.Label({
+      style_class: 'headerID'
+    });
+    label.set_text('(' + data['symbol'] + ') #' + data['cmc_rank']);
+    right.add(label);
+
+    row.add(left);
+    row.add(right);
+
+    return row;
+  },
+
+  addPrice: function(price) {
+    var row = new St.BoxLayout({
+      vertical: false,
+      width: WIDTH - PADDING,
+      style_class: 'row'
+    });
+    var center = new St.BoxLayout({
+      vertical: true,
+      width: WIDTH - PADDING,
+      style_class: 'containerPrice'
+    });
+
+    this.priceLabel = new St.Label();
+    this.priceLabel.set_text(this.getFormattedPrice(price));
+
+    center.add(this.priceLabel);
+    row.add(center);
+
+    return row;
+  },
+
+  addChange: function(title, changeLabel, changeValue) {
+    var row = new St.BoxLayout({
+      vertical: false,
+      width: WIDTH - PADDING,
+      style_class: 'row containerCenter'
+    });
+
+    var left = new St.BoxLayout({
+      vertical: true,
+      width: 110,
+      style_class: 'left'
+    });
+    var right = new St.BoxLayout({
+      vertical: true,
+      width: 100,
+      style_class: 'right'
+    });
+
+    var label = new St.Label();
+    label.set_text(title);
+    left.add(label);
+
+    this.setChangeData(changeLabel, changeValue);
+    right.add(changeLabel);
+
+    row.add(left);
+    row.add(right);
+
+    return row;
+  },
+
+  setChangeData: function(label, num) {
+    num = parseFloat(num);
+
+    var cls = 'green';
+    if (num < 0) {
+      cls = 'red';
+    }
 
-        return price.toLocaleString(undefined, options);
+    label.style_class = cls;
+    label.set_text(num.toLocaleString(undefined, {
+      style: 'decimal',
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2
+    }) + '%');
+  },
+
+  addLastUpdated: function(date) {
+    var row = new St.BoxLayout({
+      vertical: false,
+      width: WIDTH - PADDING,
+      style_class: 'row'
+    });
+    var right = new St.BoxLayout({
+      vertical: true,
+      width: WIDTH - PADDING,
+      style_class: 'lastUpdated'
+    });
+
+    date = new Date(date);
+    this.lastUpdatedLabel = new St.Label();
+    this.lastUpdatedLabel.set_text(date.toLocaleString());
+
+    right.add(this.lastUpdatedLabel);
+    row.add(right);
+
+    return row;
+  },W
```

Comment:
Typo here prevents the desklet from loading.

```
[cryptocoins@pbojan]: missing : after property id
[cryptocoins@pbojan]: Error importing desklet.js from cryptocoins@pbojan
```

Reply:
Yes, that should not be there. :-D